traveler-swift-core/TravelerKit/Models/BookingSearchParameters.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  BookingSearchParameters.swift
+//  TravelerKit
+//
+//  Created by Omar Padierna on 2019-10-04.
+//  Copyright © 2019 Ata Namvari. All rights reserved.
+//
+
+import Foundation
+/// The search parameters that conform to a `BookingQuery`
+struct BookingSearchParameters: Decodable {
+    /// Text for text-based queries
+    public let text: String?
+    ///  Range of prices for items
+    public let range: PriceRange?
+    /// Item category
+    public let categories: [CatalogItemCategory]?
```

Comment:
Are we sure optional is good here?

Reply:
Yes, search parameters are all optional for BookingQuery. The only requirement is that at least one is set

Reply:
same thinghere

Reply:
Booking search parameters is a response from the backend and it can be optional.

Reply:
you can make it non-optional as part of your decoding

Reply:
again semantics are important here

-